### ISSUE TITLE
Diff is (sometimes) not updated even after closing and re-opening

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab-git#936]
## Description

After certain sequence of events that I cannot reproduce yet the notebook diff stops updating. The old/partial/incorrect (not sure which) diff is shown instead. No error is shown to the user.

## Reproduce

In the diff there are SVG outputs and code cells.

I have not found a way to reproduce this yet. Roughly it could be something like:
- modify a tracked notebook
- added to staging area
- diff
- close diff
- modify again
- diff
- see the old/partial/incorrect diff
- see in network requests in developer tools that `diffnotebook?` correctly downloads the updated diff but it is not being displayed
- see captured nbdime error:

```
Failed to init notebook diff view: Error: Invalid arguments to createDirectStringDiffModel(). Both base and remote cannot be equal!
```

When I tried to diff that notebook using nbdime button directly:

```js
Uncaught (in promise) Error: Invalid arguments to createDirectStringDiffModel(). Both base and remote cannot be equal!
    at Object.t.createDirectStringDiffModel (568.aa5ce6fc20353381fe5b.js?v=aa5ce6fc20353381fe5b:1)
    at stringify (568.aa5ce6fc20353381fe5b.js?v=aa5ce6fc20353381fe5b:1)
    at stringify (568.aa5ce6fc20353381fe5b.js?v=aa5ce6fc20353381fe5b:1)
    at m.createOutputTextView (568.aa5ce6fc20353381fe5b.js?v=aa5ce6fc20353381fe5b:1)
    at m.createView (568.aa5ce6fc20353381fe5b.js?v=aa5ce6fc20353381fe5b:1)
    at new m (568.aa5ce6fc20353381fe5b.js?v=aa5ce6fc20353381fe5b:1)
    at Function.createView (568.aa5ce6fc20353381fe5b.js?v=aa5ce6fc20353381fe5b:1)
    at h.init (568.aa5ce6fc20353381fe5b.js?v=aa5ce6fc20353381fe5b:1)
    at new h (568.aa5ce6fc20353381fe5b.js?v=aa5ce6fc20353381fe5b:1)
    at 568.aa5ce6fc20353381fe5b.js?v=aa5ce6fc20353381fe5b:1
```

## Expected behavior

The diff should get updated; if diff cannot get updated to the new one full an error should be shown to the users to that:
- they are not shown incorrect diff (which is misleading and dangerous)
- they can report this here with more details so we can solve it

## Context

JupyterLab: 3.0.15

- Python package version: 0.30
- Git version: 2.31
- Operating System and its version: Ubuntu 21.04

**Comment by fcollonval** [comment]
Thanks @krassowski 

This one is gonna be tricky and sounds like a nbdime issue from the error message you are reporting. Let's wait if you succeed to reproduce the case.

**Comment by krassowski** [comment]
When it gets thrown both of the `createDirectStringDiffModel` are null:

![Screenshot from 2021-05-19 12-40-21](https://user-images.githubusercontent.com/5832902/118806815-6d6b6e00-b89f-11eb-9a4a-61c26b8c952b.png)

so to understand this we need to go a level up to `createOutputTextView`; this seems to be happening on an SVG diff:

![Screenshot from 2021-05-19 12-54-36](https://user-images.githubusercontent.com/5832902/118808438-6e9d9a80-b8a1-11eb-8dc6-deec42b22d51.png)

![Screenshot from 2021-05-19 12-55-06](https://user-images.githubusercontent.com/5832902/118808470-76f5d580-b8a1-11eb-8759-dff8281ac164.png)

BTW, Is there a way to enable source map for extensions easily (without installing the dev version)? It seems to work for core packages.

**Comment by krassowski** [comment]
I guess jupyterlab-git could do a better job with handling the errors for now (i.e. present this kind of errors to the user).

**Comment by krassowski** [comment]
Also tried to repro on binder but it uses 3.0.7 (for which isolated cells are broken). Would it be ok to pin JupyterLab on binder to a newer version?

**Comment by fcollonval** [comment]
> Also tried to repro on binder but it uses 3.0.7 (for which isolated cells are broken). Would it be ok to pin JupyterLab on binder to a newer version?

Sure - go ahead to push a PR; maybe something like "jupyterlab>3.0.7" in case we forget to update it later.

**Comment by fcollonval** [comment]
> Is there a way to enable source map for extensions easily (without installing the dev version)? It seems to work for core packages.

With federated module, I think the only possibility is to install nbdime in dev mode as maps are not part of the distributed package files.

**Comment by krassowski** [comment]
I managed to narrow this down and opened https://github.com/jupyter/nbdime/issues/592 with 1-cell notebook with stripped source and output. It is extremely cryptic - if I remove more of the unchanged `<symbol>` tags from the SVG source it no longer leads to an error but I could not find any regularity as to what conditions this error.

**Comment by fcollonval** [comment]
Thanks a lot @krassowski to narrow it down and produce a reproducible case. Let's wait on @vidartf.

**Comment by vidartf** [comment]
Fixed in https://github.com/jupyter/nbdime/pull/593, released as 3.1.0.

**Comment by fcollonval** [comment]
Thanks @vidartf - closing this one.
